util/pace/util/initialization/allocator.py

```diff
@@ -91,9 +101,11 @@ def from_array(
         That numpy array must correspond to the correct shape and extent
         for the given dims.
         """
-        base = self.empty(dims=dims, units=units, dtype=data.dtype)
-        base.data[:] = base.np.asarray(data)
-        return base
+        # TODO: Replace this once aligned_index fix is included.
+        quantity_data = self._numpy.from_array(
+            data, data.dtype, aligned_index=[0] * len(data.shape)
+        )
+        return Quantity(data=quantity_data, dims=dims, units=units)
```

Comment:
This is a rather large change, but something like this is required because `base.data[:] = base.np.asarray(data)` doesn't work.

Reply:
This looks dangerous, will slack to clarify.

Reply:
I think you're missing a `gt4py_backend` argument here, as well as origin and extent. This is normally handled automatically by `self.empty`. I'm also not sure whether the resulting data will or won't be optimized for the gt4py backend.

Why does `base.data[:] = base.np.asarray(data)` not work?

Reply:
Just tried it. This happens:
```     
        base = self.empty(dims=dims, units=units, dtype=data.dtype)
>       base.data[:] = base.np.asarray(data)
E       ValueError: could not broadcast input array from shape (19,19,80) into shape (19,19,3)
```

Reply:
That's a fairly serious problem worth debugging. Why was `base` initialized by `self.empty` with a z-axis length of 3? That shouldn't happen.

Reply:
Thanks for helping resolve this. We now have the arrays clipped to the correct shape.